R/get-osmdata.R

```diff
@@ -745,6 +746,46 @@ xml_to_df <- function (doc, stringsAsFactors = FALSE) {
 }
 
 
+xml_to_df_cpp <- function (doc, stringsAsFactors = FALSE) {
+
+    res <- rcpp_osmdata_df (paste0 (doc))
+    if (nrow (res$points_kv) > 0L) {
+        res$points_kv$osm_type <- "node"
+    }
+    if (nrow (res$ways_kv) > 0L) {
+        res$ways_kv$osm_type <- "way"
+    }
+    if (nrow (res$rels_kv) > 0L) {
+        res$rels_kv$osm_type <- "relation"
+    }
+
+    nms <- sort (unique (unlist (lapply (res, names))))
+    nms1 <- c ("osm_type", "osm_id", "name")
```

Comment:
Oh, the suggested change is wrong. The intersection should have `nms` as a second parameter because the resulting order of the result follows the first parameter.